### DISCUSSION TITLE
Error when creating new table

### DISCUSSION THREAD

**Discussion by TheBoatyMcBoatFace** [dbeaver/dbeaver#12055] [Q&A] [answered]
dBeaver is giving me errors when I try to create a new table

#### System information: 
- Mac OS Big Sur 11.2.3
- DBeaver version: Version 21.0.2.202104042040
- Current install via Homebrew

#### Connection specification:
- Localhost MariaDB
- MySql 10.5.9
- No tunnels or proxy, localhost

#### Describe the problem you're observing:

When I try to create a new table, I get an error. I have been creating tables all day with no issue. I can not think of any specific action that caused this problem to arise. I have tested trying to create a table via another program and it worked. I can not get dBeaver to make any more tables without an error. 

##### Problem Progression
1. Test creating table via another program (Sequel Ace)
2. Test -->  **`Success`**
3. Sequel Ace sucks except for loading data, need dBeaver
4. Uninstall dBeaver via Applications folder
5. Reinstall dBeaver via dmg
6. Test --> **`Failure`**
7. Uninstall dBeaver
8. Reinstall via Homebrew
9. Test -->  **`Failure`**
10.Uninstall via Homebrew
11. Reinstall via Homebrew
12. Update/upgrade/clean/autoremove Homebrew
13.Test -->  **`Failure`**
14.Restart computer
15.Test -->  **`Failure`**
16. Search google/github --> **`No luck`**
17. Throw expensive keyboard at wall
18. Open ticket on Github
19. Figure out how to tell wife there is a damaged wall and ruined keyboard

#### Steps to reproduce:

1. Click Create Table
2. dBeaver gives below code to create table

```MYSQL
CREATE TABLE Testy.McTestFace (

)
ENGINE=InnoDB
DEFAULT CHARSET=utf8mb4
COLLATE=utf8mb4_general_ci;
```
3. Press Create
4. Get Error

#### Error Window

[![Screen-Shot-2021-04-08-at-3-08-06-PM.png](https://i.postimg.cc/jqyFPj13/Screen-Shot-2021-04-08-at-3-08-06-PM.png)](https://postimg.cc/t7CkH9pF)

##### Top of Error Window
```
You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
ENGINE=InnoDB
DEFAULT CHARSET=utf8mb4
COLLATE=utf8mb4_general_ci' at line 3
```

##### More Details Window

```
SQL Error [1064] [42000]: (conn=24) You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
  (conn=24) You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
  (conn=24) You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
    You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
    You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
      You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
ENGINE=InnoDB
DEFAULT CHARSET=utf8mb4
COLLATE=utf8mb4_general_ci' at line 3
      You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
ENGINE=InnoDB
DEFAULT CHARSET=utf8mb4
COLLATE=utf8mb4_general_ci' at line 3
```
#### Log Files

```java
eclipse.buildId=unknown
java.version=11.0.8
java.vendor=AdoptOpenJDK
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_US
Framework arguments:  -keyring /Users/bentleyhensel/.eclipse_keyring
Command-line arguments:  -os macosx -ws cocoa -arch x86_64 -keyring /Users/bentleyhensel/.eclipse_keyring

org.jkiss.dbeaver.model
Error
Thu Apr 08 14:52:51 EDT 2021
You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
ENGINE=InnoDB
DEFAULT CHARSET=utf8mb4
COLLATE=utf8mb4_general_ci' at line 3
```

**Comment by flambchap** [comment, marked as answer]
Hello @TheBoatyMcBoatFace. AFAIK, it's impossible to create a table without specifying at least one column in MariaDB. Please try to change your create statement and everything should be fine.
